src/helm/benchmark/scenarios/melt_scenarios.py

```diff
@@ -343,3 +351,947 @@ def __init__(self, **kwargs):
+        "sợ": ["sợ"],
+        "lạnh": ["lạnh", "lạnh buốt", "mát mẻ"],
+        "nóng": ["nóng", "ấm"],
+        "thông minh": ["thông minh", "tài giỏi", "khôn", "sáng trí"],
+        "sạch": ["sạch", "ngăn nắp"],
+        "nhỏ": ["nhỏ", "bé", "tí nị"],
+        "to": ["to", "khổng lồ", "bự", "lớn"],
+        "tốt": ["tốt", "tử tế", "tốt bụng"],
+        "đẹp": ["đẹp", "xinh"],
+        "đỏ": ["đỏ", "đỏ thẫm"],
+        "xanh dương": ["xanh dương", "xanh lam"],
+        "xanh lục": ["xanh lục", "xanh lá cây"],
+        "tím": ["tím", "tím than"],
+        "chán": ["chán", "đần"],
+        "cũ": ["cũ", "xưa", "cổ"],
+        "mạnh": ["mạnh", "mạnh mẽ", "cơ bắp"],
+        "yếu": ["yếu", "yếu đuối", "mỏng manh"],
+        "nhanh": ["nhanh", "mau"],
+        "chậm": ["chậm", "chậm chạp"],
+        "xấu": ["xấu", "xấu xa", "ác", "độc ác"],
+        "hạnh phúc": ["hạnh phúc", "hân hoan", "vui mừng", "vui vẻ"],
+        "tròn": ["tròn", "hình tròn", "hình cầu"],
+    }
+    # Remove any keys which duplicate subitems
+    new_attribute_groups: Dict[str, List[str]] = copy(attribute_groups)
+    for general_attribute, specific_attributes in attribute_groups.items():
+        for specific_attribute in specific_attributes:
+            if (general_attribute != specific_attribute) and (specific_attribute in attribute_groups):
+                del new_attribute_groups[specific_attribute]
+
+    return new_attribute_groups, subjects
+
+
+def generate_rules(
+    attribute_groups: Dict[str, List[str]],
+    subject: str,
+    subject_category: str,
+    max_rules: int = 5,
+    specific_category: bool = False,
+) -> List[MELTLanguageRule]:
+    """Generates a random set of rules about a subject as dictionaries,
+    given a list of potential attributes and the category (e.g. người) of the subject (e.g. An)
+
+    These rules are guaranteed to not contradict one another, and attributes implied by a single rule will
+    not imply any attributes in any other rules (i.e. there is only a single step of reasoning).
+    """
+    attributes_shuffled = list(attribute_groups.keys()).copy()
+    random.shuffle(attributes_shuffled)
+    rules: List[MELTLanguageRule] = []
+
+    while len(attributes_shuffled) > 2 and len(rules) < max_rules:
+        rule_subject = subject if specific_category else random.choice([subject_category, subject])
+        n_rule_attributes = random.randint(2, 3)
+        rule_attributes, attributes_shuffled = (
+            attributes_shuffled[:n_rule_attributes],
+            attributes_shuffled[n_rule_attributes:],
+        )
+        rules.append(
+            MELTLanguageRule(
+                subject=rule_subject,
+                subject_category=subject_category,
+                specifier_type="một",
+                condition=rule_attributes[:-1],
+                condition_conjunction=random.choice(["và", "hoặc"]),
+                consequent=rule_attributes[-1],
+            )
+        )
+    return rules
+
+
+def generate_test(
```

Comment:
I'm generally not reviewing for correctness, but let me know if there is any specific program logic that you would like me to review.